### PR TITLE
5.0.0.rc

### PR DESCRIPTION
**Pull request by mamoonraja**

### CODE REVIEW

File: Scripts/Services/Assistant/V1/Model/DialogNode.cs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
-* (C) Copyright IBM Corp. 2018, 2020.
+* (C) Copyright IBM Corp. 2019, 2020.
```
Comment (mediumTaj) [inline Scripts/Services/Assistant/V1/Model/DialogNode.cs:2]: strange that these are being changed

File: Scripts/Services/Assistant/V1/Model/RuntimeResponseGeneric.cs [hunk 2]
```diff
@@ -35,26 +54,6 @@ public class ResponseTypeValue
...
-            /// </summary>
-            public const string OPTION = "option";
-            /// <summary>
-            /// Constant CONNECT_TO_AGENT for connect_to_agent
-            /// </summary>
-            public const string CONNECT_TO_AGENT = "connect_to_agent";
-            /// <summary>
-            /// Constant SUGGESTION for suggestion
-            /// </summary>
-            public const string SUGGESTION = "suggestion";
```
Comment (mediumTaj) [inline Scripts/Services/Assistant/V1/Model/RuntimeResponseGeneric.cs:57]: Are these being replaced somewhere else? or maybe not needed as its a response?

File: Scripts/Services/VisualRecognition/V4/VisualRecognitionService.cs [hunk 3]
```diff
@@ -72,6 +72,7 @@ public bool DisableSslVerification
         /// <param name="authenticator">The service authenticator.</param>
         public VisualRecognitionService(string versionDate, Authenticator authenticator) : base(versionDate, authenticator, serviceId)
         {
+            Log.Warning("VisualRecognition Deprecation Warning:", "On 1 December 2021, Visual Recognition will no longer be available. For more information, see https://github.com/watson-developer-cloud/unity-sdk/tree/master#visual-recognition-deprecation.");
```
Comment (mediumTaj) [inline Scripts/Services/VisualRecognition/V4/VisualRecognitionService.cs:75]: does `[Obsolete("message"]` work on the class? see https://github.com/watson-developer-cloud/dotnet-standard-sdk/blob/e8f160aac7c3d84e2279e30d4fa092ca1591a90b/src/IBM.Watson.PersonalityInsights.v3/PersonalityInsightsService.cs

**Review by mediumTaj — CHANGES_REQUESTED** [review]
Looks good overall mamoon, just a couple of comments. Also I don't see personality insights deprecation in this pr, has it been added already?

File: Scripts/Services/Assistant/V1/Model/DialogNode.cs (hunk 1 above)
Comment (mamoonraja) [reply Scripts/Services/Assistant/V1/Model/DialogNode.cs:2]: I am not sure how this is happening. Looking into it

File: Scripts/Services/VisualRecognition/V4/VisualRecognitionService.cs (hunk 3 above)
Comment (mamoonraja) [reply Scripts/Services/VisualRecognition/V4/VisualRecognitionService.cs:75]: Let me try

File: Scripts/Services/Assistant/V1/Model/DialogNode.cs (hunk 1 above)
Comment (mediumTaj) [reply Scripts/Services/Assistant/V1/Model/DialogNode.cs:2]: strangely the command in lint-copyrights gives 2019
```
git log --pretty=format:"%ad" --follow -M95% --date=format:%Y Scripts/Services/Assistant/V1/AssistantService.cs
```

Comment (mamoonraja) [reply Scripts/Services/Assistant/V1/Model/DialogNode.cs:2]: In this case, it is correct: https://github.com/watson-developer-cloud/unity-sdk/commits/52f0cf940865fc28cbe14833c69bed8293149a1a/Scripts/Services/Assistant/V1/Model/DialogNode.cs

Theses got refactored in 2019: https://github.com/watson-developer-cloud/unity-sdk/tree/v2.13.2 in 2.14.0

**Review by mediumTaj — APPROVED** [review]
👍 looks good!

**Review by mediumTaj — CHANGES_REQUESTED** [review]
Need to merge in https://github.com/watson-developer-cloud/unity-sdk/pull/663 first and update tests

File: Tests/DiscoveryV2IntegrationTests.cs [hunk 4]
```diff
@@ -50,6 +50,8 @@ public class DiscoveryV2IntegrationTests
         public void OneTimeSetup()
         {
             LogSystem.InstallDefaultReactors();
+            analyzeDocumentFile = Application.dataPath + "/Watson/Tests/TestData/DiscoveryV1/exampleConfigurationData.json";
```
Comment (mediumTaj) [inline Tests/DiscoveryV2IntegrationTests.cs:53]: Are these running? I think we can exclude disco v2 tests as we don't have a persistent instance

**Review by mediumTaj — APPROVED** [review]
👍 looks much better!

**Review by mediumTaj — APPROVED** [review]

**Pull request merged**